resource-service/web/src/test/java/com/epam/resource/PostResourceStepdefs.java

```diff
@@ -0,0 +1,99 @@
+package com.epam.resource;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.epam.resource.entity.ResourceEntity;
+import com.epam.resource.repository.ResourceRepository;
+import com.epam.resource.util.Id;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import io.cucumber.messages.internal.com.fasterxml.jackson.annotation.JsonIgnore;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Tag;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+@Tag("component-test")
+public class PostResourceStepdefs {
+    @Autowired
+    private ResourceRepository resourceRepository;
+
+    @Autowired
+    private AmazonS3 amazonS3;
+
+    @Autowired
+    private TestRestTemplate template;
+
+    @Value("${amazon.s3.bucket-name}")
+    private String bucketName;
+
+    @LocalServerPort
+    private int port;
+
+    @JsonIgnore
+    private MultipartFile fileToUpload;
+    private Id actual;
```

Comment:
It seems like the value can be shared between step defs, so, please consider in such cases to use ThreadLocal to wrap the field in case if it should be shared. It is necessary because you may want to parallel cucumber tests. Also, it is a good practice to create a separate class ("TestContext") which will contain all values that should be shared between step definitions.